lib/spack/spack/schema/packages.py

```diff
@@ -97,3 +111,33 @@
     'additionalProperties': False,
     'properties': properties,
 }
+
+
+def update(data):
+    """Update in-place the data to remove deprecated properties.
+
+    Args:
+        data (dict): dictionary to be updated
+
+    Returns:
+        True if data was changed, False otherwise
+    """
```

Comment:
This PR adds a new command:
```console
$ spack config update
```
to update configuration files when there's a change in the schema. This made me wonder if we want to start versioning the schema for configuration files as we do for the DB. Thoughts?

Reply:
I agree that the schemas should have versions.